File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/entity/LoginLog.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("login_log")
public class LoginLog implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;

    @TableId(value = "log_id", type = IdType.AUTO)
    private Integer logId;

    @TableField("username")
    private String username;

    @TableField("login_ip_address")
    private String loginIpAddress;

    @TableField("login_time")
    private LocalDateTime loginTime;

    @TableField("login_result")
    private String loginResult;

    @TableField("browser_type")
    private String browserType;

    @TableField("os_version")
    private String osVersion;

    @TableField("remarks")
    private String remarks;
}

File: final_asssignment_backend_go/project/internal/domain/login_log.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// LoginLog 表示 login_log 表的实体
type LoginLog struct {
	LogID          int            `gorm:"column:log_id;primaryKey;autoIncrement" json:"log_id"`
	Username       string         `gorm:"column:username" json:"username"`
	LoginIPAddress string         `gorm:"column:login_ip_address" json:"login_ip_address"`
	LoginTime      time.Time      `gorm:"column:login_time" json:"login_time"`
	LoginResult    string         `gorm:"column:login_result" json:"login_result"`
	BrowserType    string         `gorm:"column:browser_type" json:"browser_type"`
	OSVersion      string         `gorm:"column:os_version" json:"os_version"`
	Remarks        string         `gorm:"column:remarks" json:"remarks"`
	DeletedAt      gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (LoginLog) TableName() string {
	return "login_log"
}
